src/constraint_system/composer.rs

```diff
@@ -206,7 +229,9 @@ impl StandardComposer {
         self.q_fixed_group_add.push(BlsScalar::zero());
         self.q_variable_group_add.push(BlsScalar::zero());
 
-        self.public_inputs.push(pi);
+        if let Some(pi) = pi {
+            assert!(self.public_inputs_sparse_store.insert(self.n, pi).is_none());
```

Comment:
Unnecessary assertion, check above

Reply:
Same comment I made in the previous one.